Xplat/src/main/java/vazkii/botania/common/helper/StructureHelper.java

```diff
@@ -0,0 +1,17 @@
+package vazkii.botania.common.helper;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.level.levelgen.structure.Structure;
+
+public class StructureHelper {
+	public static Structure getStructure(ServerLevel level, ResourceLocation key) {
+		return level.registryAccess().registryOrThrow(Registry.STRUCTURE_REGISTRY).get(key);
+	}
+
+	public static boolean isInStructureBounds(ServerLevel level, BlockPos pos, Structure structure) {
+		return structure == null ? false : level.structureManager().getStructureAt(pos, structure).isValid();
```

Comment:
```suggestion
		return structure != null && level.structureManager().getStructureAt(pos, structure).isValid();
```

Reply:
taken straight from vanilla code, but I guess that works too

Reply:
If there's vanilla code, why not call that directly?

Reply:
The vanilla code is part of a method that's specific to nether fortresses